libs/sr-opt/src/optimize.ts

```diff
@@ -0,0 +1,83 @@
+import { detach } from '@genshin-optimizer/pando'
+import {
+  allRelicSetKeys,
+  type RelicSlotKey,
+} from '@genshin-optimizer/sr-consts'
+import type { ICachedRelic } from '@genshin-optimizer/sr-db'
+import type { Calculator, Read, Tag } from '@genshin-optimizer/sr-formula'
+import type { ParentCommand, ParentMessage } from './parentWorker'
+
+export interface BuildResult {
+  value: number
+  ids: Record<RelicSlotKey, string>
+}
+
+export interface ProgressResult {
+  numBuilds: number
+  numBuildsCompute: number
+  waitingForResults?: boolean | undefined
+}
+
+export const MAX_BUILDS = 2_000_000
+
+export async function optimize(
+  calc: Calculator,
+  optTarget: Read,
+  relicsBySlot: Record<RelicSlotKey, ICachedRelic[]>,
+  numWorkers: number,
+  setProgress: (progress: ProgressResult) => void
+) {
+  // Spawn a parent worker to compile nodes, split/filter relics and spawn child workers for calculating results
+  const worker = new Worker(new URL('./parentWorker.ts', import.meta.url), {
+    type: 'module',
+  })
+
+  // Wait for parent worker to report complet
+  const buildResults = await new Promise<BuildResult[]>((res, rej) => {
+    worker.onmessage = ({ data }: MessageEvent<ParentMessage>) => {
+      switch (data.resultType) {
+        case 'progress':
+          setProgress({
+            numBuilds: data.numBuilds,
+            numBuildsCompute: data.numBuildsComputed,
+            waitingForResults: data.sendingResults,
```

Comment:
Maybe we can match the variable names, e.g. `computing`, so we can just send `data` over?